src/rust/options/src/pants_ng_flags.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use crate::fromfile::FromfileExpander;
+use crate::id::NameTransform;
+use crate::parse::{ParseError, Parseable};
+use crate::{DictEdit, ListEdit, OptionsSource};
+use crate::{OptionId, Scope};
+use std::any::Any;
+use std::collections::HashMap;
+
+// This properly belongs in pants_ng/options/flags.rs, but it depends on a lot of non-public
```

Comment:
Reviewers: If you prefer I can move this under `pants_ng/options`, and change various entities' visibility to `pub`. I think there are ~6 of them.